[CI] AllSupportedFieldsIT class failing

**Build Scans:**
- [elasticsearch-intake #38309 / 9.3.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ic54a7xo7xgh6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v9.3.4#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.AllSupportedFieldsIT" -Dtests.method="testFetchAll {pref=null mode=lookup}" -Dtests.seed=3C165C521B4110E3 -Dtests.bwc=true -Dtests.locale=ia -Dtests.timezone=US/Alaska -Druntime.java=26
```

**Applicable branches:**
9.4

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.AllSupportedFieldsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.xpack.esql.qa.mixed.AllSupportedFieldsIT'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: partial results: {took=84, completion_time_in_millis=1776996336652, _clusters={running=0, total=1, details={(local)={_shards={total=3, failed=0, successful=3, skipped=0}, took=84, indices=lookup*, failures=[{reason={reason=index [lookup_test-cluster-0] has no active shard copy, type=no_shard_available_action_exception}, index=null, shard=-1}], status=partial}}, failed=0, partial=1, successful=0, skipped=0}, is_partial=true, columns=[{name=@timestamp, type=date}, {name=f_aggregate_metric_double, type=aggregate_metric_double}, {name=f_boolean, type=boolean}, {name=f_byte, type=integer}, {name=f_counter_double, type=double}, {name=f_counter_integer, type=integer}, {name=f_counter_long, type=long}, {name=f_date_nanos, type=date_nanos}, {name=f_date_range, type=unsupported, original_types=[date_range], suggested_cast=keyword}, {name=f_dense_vector, type=dense_vector}, {name=f_double, type=double}, {name=f_exponential_histogram, type=exponential_histogram}, {name=f_
[truncated]
```

**Issue Reasons:**
- [9.4] 1 consecutive failures in class org.elasticsearch.xpack.esql.qa.mixed.AllSupportedFieldsIT
- [9.4] 1 consecutive failures in step 9.3.4_bwc-snapshots
- [9.4] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
[2026-04-23T18:05:35,036][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-0] fatal error in thread [elasticsearch[test-cluster-0][search][T#4]], exiting java.lang.AssertionError: Unknown NamedWriteable [org.elasticsearch.compute.operator.Operator$Status][streaming_lookup]	
	at org.elasticsearch.server@9.3.4-SNAPSHOT/org.elasticsearch.common.io.stream.NamedWriteableRegistry.throwOnUnknownWritable(NamedWriteableRegistry.java:150)
```

In the same build scan, another test fails due to node dying from the unknown join-related named writeable:
```
[2026-04-24T12:13:54,058][ERROR][o.e.ExceptionsHelper     ] [test-cluster-2] fatal error java.lang.AssertionError: Unknown NamedWriteable [org.elasticsearch.compute.operator.Operator$Status][streaming_lookup]
```
That happens in the test with reproducer
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v9.3.4#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT" -Dtests.method="test {csv-spec:lookup-join.halfFloatJoinShort}" -Dtests.seed=3C165C521B4110E3 -Dtests.bwc=true -Dtests.locale=ff-Latn-GM -Dtests.timezone=Antarctica/DumontDUrville -Druntime.java=26	
```

--- a/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/lookup/LookupQueryOperator.java
+++ b/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/lookup/LookupQueryOperator.java
@@ -334,6 +334,8 @@ public final class LookupQueryOperator implements Operator {
             Status::new
         );
 
+        private static final TransportVersion ESQL_STREAMING_LOOKUP_JOIN = TransportVersion.fromName("esql_streaming_lookup_join");
+
         private final int pagesReceived;
         private final int pagesEmitted;
         private final long rowsReceived;
@@ -424,7 +426,7 @@ public final class LookupQueryOperator implements Operator {
 
         @Override
         public TransportVersion getMinimalSupportedVersion() {
-            return TransportVersion.current();
+            return ESQL_STREAMING_LOOKUP_JOIN;
         }
     }
 }

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/StreamingLookupFromIndexOperator.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/StreamingLookupFromIndexOperator.java
@@ -836,7 +836,7 @@ public class StreamingLookupFromIndexOperator implements Operator {
 
         @Override
         public TransportVersion getMinimalSupportedVersion() {
-            return TransportVersion.current();
+            return ESQL_LOOKUP_PLAN_STRING;
         }
 
         @Override

